styling and templating updated

dejmail opened the pull request:
Have updated the styling and templating used so that the content is more in line with that which is currently being used. I have not updated the "time required" per lesson because I have not held the lessons myself, and so do not know how long to apportion to each.

katyhuff commented:
Thanks for all this work. Unfortunately, this branch has conflicts with the main branch. Can you please resolve them?

dejmail commented:
Ok, sorry to be a pain. Do you have any tips about how I do this..seems intuitive that it should be simple but it's not quite clear. Do I rebase my repo with yours to see the difference ?

katyhuff commented:
@dejmail see my comment here: https://github.com/katyhuff/python-testing/issues/8#issuecomment-280359801

katyhuff commented on _includes/syllabus.html:
```diff
@@ -0,0 +1,75 @@
+  <tr>
+    <td class="col-md-1"></td>
+    <td class="col-md-3"><a href="{{ page.root }}/setup">Setup</a></td>
+    <td class="col-md-7">Dowload files used on the lesson.</td>
```
Spelling. Also, not quite proper english. Perhaps "Download files used in the lesson." Also, the link leads to a completely blank setup page?

dejmail replied:
Yes, sorry, my english is getting worse and worse. I blame it on a typo. Will create the setup page.

dejmail commented:
Think it's ok now ?

katyhuff commented on .gitignore:
```diff
@@ -4,7 +4,10 @@ FAQ.html
 CONDUCT.html
+*.orig
 *~
-*.pyc
```
shouldn't we always ignore .pyc?

dejmail replied:
Indeed, did I do that ? Will put it back.

katyhuff commented on CONDUCT.md:
```diff
@@ -1,13 +1,44 @@
-# Contributor Code of Conduct
+---
```
Most of the changes in this file are linebreaks. the only new functionality is the first 5 lines. Perhaps the commit shouldn't include all these extra non substantive line changes. It's fine either way, but not ideal this way... it just creates history where there is none.

dejmail replied:
I agree. If my memory serves me correctly, they were put in to stop the Jekyll compiler complaining about lines that were too long. I can remove them though, especially where it is just me deleting whitespace.

dejmail replied:
I've removed the extra linebreaks, though I think they were all added by Greg.

katyhuff commented on _config.yml:
```diff
@@ -0,0 +1,70 @@
+
+# Magic to make URLs resolve both locally and on GitHub.
+# See https://help.github.com/articles/repository-metadata-on-github-pages/.
+repository: <USERNAME>/<PROJECT>
```
should this not be filled out with the reality? katyhuff/python-testing ?

dejmail replied:
Yes, it should. Will change it.

katyhuff commented on _config.yml:
```diff
@@ -0,0 +1,70 @@
+
+# Contact email address.
+email: lessons@software-carpentry.org
+contact: lessons@software-carpentry.org 
```
I'm the maintainer but I don't receive emails sent to lessons@software-carpentry.org. Perhaps this should be katyhuff@gmail.com ?

dejmail replied:
Sure, will change it.

katyhuff commented:
This is super long, so it's taking me a while to review. Sorry for the delay. Will need  a few more days.

dejmail commented:
It's really not a problem, take your time.

katyhuff commented on LICENSE.md:
```diff
@@ -1,13 +1,15 @@
-the [Creative Commons Attribution license][cc-by-human]. The following
-is a human-readable summary of (and not a substitute for) the [full
-legal text of the CC BY 4.0 license][cc-by-legal].
+All Software Carpentry and Data Carpentry instructional material is
```
only changes here are linebreaks..

dejmail replied:
Greg Wilson put these in, can we ignore that they were added ?

katyhuff commented on _episodes/01-basics.md:
```diff
@@ -1,13 +1,20 @@
+teaching: 5
+exercises: 0
+questions:
+- "Why would you not want to test?"
```
Huh?

dejmail replied:
Each pages starts with a question in the newer template that is hopefully answered by the end of the section. Quite likely not the best question to start with-.

katyhuff commented on _episodes/#01-basics.md#:
```diff
@@ -0,0 +1,57 @@
+teaching: 5
+exercises: 0
+questions:
+- "Why would you not want to test?"
```
I don't understand why this shouldn't be "Why test"

dejmail replied:
Changed to "Why test?"

katyhuff commented on _episodes/#01-basics.md#:
```diff
@@ -0,0 +1,57 @@
+- "Understand that testing has many forms."
+keypoints:
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
+- "Tests should be written at the same time as the code they are testing is written."
```
I don't necessarily think this is true. Tests are, arguably, more effective if written beforehand. And, many tests must, by their nature, be written after. Where is this statement coming from?

dejmail replied:
Have expanded sentence to include your suggestions.

katyhuff commented on _episodes/#01-basics.md#:
```diff
@@ -0,0 +1,57 @@
+- "Understand the place of testing in a scientific workflow."
+- "Understand that testing has many forms."
+keypoints:
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
```
Rephrase for clarity "Tests check whether the observed result, from running the code, is what was expected ahead of time."

dejmail replied:
Done.

katyhuff commented on _episodes/#01-basics.md#:
```diff
@@ -0,0 +1,57 @@
+keypoints:
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
+- "Tests should be written at the same time as the code they are testing is written."
+- "Assertions and exceptions are li02ke alarm systems embedded in the software, guarding against exceptional bahavior."
```
li02ke is a typo.

dejmail replied:
Corrected.

katyhuff commented on _episodes/#01-basics.md#:
```diff
@@ -0,0 +1,57 @@
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
+- "Tests should be written at the same time as the code they are testing is written."
+- "Assertions and exceptions are li02ke alarm systems embedded in the software, guarding against exceptional bahavior."
+- "Unit tests try to test the smallest pieces of code possible, usually functions and methods."
```
smallest pieces -> most atomic units

katyhuff commented on _episodes/01-basics.md:
```diff
@@ -1,13 +1,20 @@
+- "Understand the place of testing in a scientific workflow."
+- "Understand that testing has many forms."
+keypoints:
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
```
same rephrasing suggested here as before.

katyhuff commented on _episodes/01-basics.md:
```diff
@@ -1,13 +1,20 @@
+- "Understand that testing has many forms."
+keypoints:
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
+- "Tests should be written at the same time as the code they are testing is written."
```
Same concern here. I don't necessarily advocate a single time for writing tests.

katyhuff commented on _episodes/01-basics.md:
```diff
@@ -1,13 +1,20 @@
+- "Tests compare that the result observed from running code is the same as what was expected ahead of time."
+- "Tests should be written at the same time as the code they are testing is written."
+- "Assertions and exceptions are like alarm systems embedded in the software, guarding against exceptional bahavior."
+- "Unit tests try to test the smallest pieces of code possible, usually functions and methods."
```
atomic units

dejmail replied:
Were you meaning "Unit tests try to test the smallest pieces of code possible, usually functions and methods." as "Unit tests try to test the smallest atomic units of code possible, usually functions and methods." ?

katyhuff commented on _episodes/02-assertions.md:
```diff
@@ -1,43 +1,54 @@
+teaching: 10
+exercises: 0
+questions:
+- "How do I make sure the value I want is the value I receive?"
```
Maybe "How can we compare observed and expected values?"

dejmail replied:
Changed.

katyhuff commented on _episodes/02-assertions.md:
```diff
@@ -48,27 +59,23 @@ Also, input checking may require decending a rabbit hole of exceptional cases.
+{: .callout}
 
-> ## Almost Equal {.challenge}
+> ## Testing similarity between variables
```
This heading seems long. Maybe "Testing Near Equality" ?

dejmail replied:
Changed.

katyhuff commented on _episodes/03-exceptions.md:
```diff
@@ -1,29 +1,34 @@
+teaching: 10
+exercises: 0
+questions:
+- "How do I capture errors that I do not anticipate?"
```
It's important that we communicate that these are runtime errors. Anyway, you have to anticipate them to correctly write an exception. I'd phrase this differently. Perhaps "How do I handle unusual behavior while the code runs?"

dejmail replied:
Changed.

katyhuff commented on _episodes/03-exceptions.md:
```diff
@@ -35,52 +40,53 @@ global scope and halts execution.
+{: .python}
 
-> ## What Else Can Go Wrong? {.challenge}
+> ## Remember
```
Why is this called Remember?

dejmail replied:
Changed back to "What else can go wrong?"

katyhuff commented on _episodes/04-units.md:
```diff
@@ -1,16 +1,22 @@
+teaching: 10
+exercises: 0
+questions:
+- "How much is a unit of code?"
```
How much -> What

dejmail replied:
Changed.

katyhuff commented on _episodes/04-units.md:
```diff
@@ -24,14 +30,15 @@ project and language to language.  A good guideline is that if the code cannot
 practically (a function is self-contained and well defined), then it is a unit. 
 
-> ## Functions are Like Paragraphs {.callout}
+> ## Note to self
```
Why is this "Note to Self" ? I don't get it. It should be called "Functions are Like Paragraphs".

dejmail replied:
Changed back to "Functions are Like Paragraphs"

katyhuff commented on _episodes/06-edges.md:
```diff
@@ -28,7 +31,7 @@ Anecdotally, it is important to test edges cases because this is where errors te
 they tend to have special code dedicated to them in the implementation.
 
-> ## Consider the Fibonacci Sequence {.callout}
+> ## Consider a Fibonacci sequence
```
There is only one Fibonacci sequence. It should remain "the Fibonacci sequence".

dejmail replied:
Indeed. Changed.

katyhuff commented on _episodes/06-edges.md:
```diff
@@ -75,23 +79,27 @@ def test_fib6():
+learned in the assertions section that this is actually quite a good idea.
 
-> ## Test for Graceful Failure {.challenge}
+> ## Implement these points
```
Why change the title here? I don't think "Implement these points" is descriptive at all.

dejmail replied:
Changed back to "Test for Graceful Failure"

katyhuff commented on _episodes/06-edges.md:
```diff
@@ -147,8 +156,9 @@ def test_edge_y():
+{: .python}
 
-> ## Write a Corner Case {.challenge}
+> ## Final steps
```
Why change this heading title? I prefer Write a Corner Case

dejmail replied:
Changed back.

katyhuff commented on _episodes/07-integration.md:
```diff
@@ -1,14 +1,16 @@
+teaching: 10
+exercises: 0
+questions:
+- "What is beyond a unit test?"
```
Maybe "How do we test more than a single unit of software?"

dejmail replied:
Changed to suggestion.

katyhuff commented on _episodes/07-integration.md:
```diff
@@ -17,7 +19,7 @@ the gears and cogs that make up the system. On their own, they can be of the hig
 needs to be put together. The gears need to fit with one another.
 
-> ## Telling The Time {.callout}
+> ## Further explanation
```
Why all of these changes of the titles? I prefer Telling the Time. It refers to the integration test as a clock made up of units. Why do you feel "Further Explanation" is better?

dejmail replied:
Not sure. I have changed it back to what you wrote.

katyhuff commented on _episodes/07-integration.md:
```diff
@@ -101,13 +104,14 @@ the expected result should be, regression tests look to the past for the
 result is taken as what was previously computed for the same inputs.
 
-> ## The Past as Truth {.callout}
+> ## Further explanation
```
Is there a reason for this change?

dejmail replied:
Nope. Will revert all those you've suggested. I honestly don't recall making them.

katyhuff commented on _episodes/08-ci.md:
```diff
@@ -1,33 +1,37 @@
+teaching: 10
+exercises: 0
+questions:
+- "How can I make sure my changes are compatable with the larger code base?"
```
No. This is about automating tests. "How can I automate running the tests on more platforms than my own?"

dejmail replied:
Changed to suggestion.

katyhuff commented on _episodes/08-ci.md:
```diff
@@ -1,33 +1,37 @@
+keypoints:
+- "Servers exist for automatically running your tests"
+- "Running the tests can be triggered by a GitHub pull request"
+- "CI allows cross platform build testing"
```
cross-platform

dejmail replied:
Corrected.

katyhuff commented on _episodes/08-ci.md:
```diff
@@ -1,33 +1,37 @@
 continuous integration system builds and checks that code.
 
-> ## Thought Experiment: Does Your Software Work on Your Colleague's Computer? {.challenge}
+> ## Example
```
If this heading has to change, at least retain the content in the title. If you must change the heading for some reason, then this section should begin "Consider the following thought experiment: Does your software work on your colleague's computer?"

dejmail replied:
Changed back.

katyhuff commented on _episodes/08-ci.md:
```diff
@@ -36,7 +40,7 @@ repository, rebuild the code, and run the tests. If you don't have a build
 compatible.
 
-> ## Let The Computers Do The Work {.callout}
+> ## It saves you time!
```
Why lose this thought about letting the computers do the work?

dejmail replied:
Changed back to what you had.

katyhuff commented on _episodes/08-ci.md:
```diff
@@ -47,56 +51,59 @@ compatible.
 
-> ## Set Up a Mean Git Repository on GitHub {.challenge}
+
+> ## First steps...
```
Why this change of title?

dejmail replied:
Changed back to what you had.

katyhuff commented on _episodes/09-tdd.md:
```diff
@@ -1,13 +1,18 @@
+- "Test driven development is a common software development technique"
+- "By writing the tests first, the function requirements are very explicit"
+- "TDD is not for everyone"
+- "TDD requires vigilance for success (cheating leads to failure)"
```
I don't like the paranthetical "(cheating leads to failure)". That's not the point.

dejmail replied:
Removed.

katyhuff commented on _episodes/09-tdd.md:
```diff
@@ -21,7 +26,7 @@ needed, you write another test and then go back and modify the function.  You
 implemented for your current needs.
 
-> ## The Big Idea {.callout}
+> ## Where it comes from
```
Where what comes from? Please give a compelling reason for this change.

dejmail replied:
Changed back to what you had.

katyhuff commented on _episodes/09-tdd.md:
```diff
@@ -32,42 +37,45 @@ implemented for your current needs.
 is too important to be an afterthought.**
 
-> ## You Do You {.callout}
+> ## Who is using it?
```
This was a nod to pop culture. The heading change doesn't make any sense. The paragraph is not at all about "who is using it
 Please give a compelling reason for this change or revert it.

dejmail replied:
Changed back to what you had.

katyhuff commented on _episodes/10-fixtures.md:
```diff
@@ -1,23 +1,24 @@
+teaching: 10
+exercises: 0
+questions:
+- "How do I create and cleanup the data I need to test the code?"
```
good

katyhuff commented on _episodes/10-fixtures.md:
```diff
@@ -1,23 +1,24 @@
 following example that could arise when comunicating with third-party programs.
 You have a function `f()` which will write a file named `yes.txt` to disk with
-the value 42 but only if a file `no.txt` does not exist. To truly test the
+the value 42 but only if a file `no.txt` does not exist. To truly test that the
```
good catch

dejmail commented:
I will go through all the comments that you made. To be honest, I am not sure where several of the changes that I submitted came from, but am happy to revert to what you suggest. I only intended to change the styling, but perhaps some of the content that you had written got mangled in the process (or perhaps Greg added it, not sure). Anyway, I apologise for that, as it was unnecessary for you to read through.

I might take me a little bit to make the changes you want, as I'm a bit snowed under with other things at the moment.

katyhuff commented:
Thanks so much! I'm sorry I was a little picky. I really am enormously
grateful for your huge effort here! Thanks so much!
Take your time (I certainly took my time reviewing it in the first place...
sorry again about that!)

On Tue, Apr 4, 2017 at 1:27 AM, Liam_Thompson <notifications@github.com>
wrote:

> I will go through all the comments that you made. To be honest, I am not
> sure where several of the changes that I submitted came from, but am happy
> to revert to what you suggest. I only intended to change the styling, but
> perhaps some of the content that you had written got mangled in the process
> (or perhaps Greg added it, not sure). Anyway, I apologise for that, as it
> was unnecessary for you to read through.
>
> I might take me a little bit to make the changes you want, as I'm a bit
> snowed under with other things at the moment.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/katyhuff/python-testing/pull/12#issuecomment-291405556>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYCq1Kp0v2_Ccp9fg23mL8X5k5LP3FTks5rseLagaJpZM4MCxRk>
> .
>



-- 
http://katyhuff.github.com

dejmail commented:
Ok. Apologies it took so long. New job and addition to family makes one short on extra time.

I think I have updated all the comments you added above and brought it up to date with other additions from your repo. 

I reverted all the headers and additions I seem to have made. The files with the line breaks, whilst unnecessary, I think can be safely ignored. They seem to have come across is Greg's merge upstream and only seem to be in the LICENSE and CONDUCT files.

The instructor notes link has nothing in it. Perhaps you might like to add something or we can leave it blank.

katyhuff commented:
This looks great!!! Thanks so much @dejmail !
